tests/unit_test/inputs/test_base_transformer.py

```diff
@@ -0,0 +1,70 @@
+import pytest
+from matchzoo.engine.base_transformer import BaseTransformer
+
+from collections import defaultdict
+import copy
+
+
+def test_base_transformer_abstract_instantiation():
+    with pytest.raises(TypeError):
+        transformer = BaseTransformer()
+        assert transformer
+
+def test_base_transformer_concrete_instantiation():
```

Comment:
This test is way too complicated. This is usually a sign of test-unfriendly implementation of the target.

Reply:
If it's a doc test, which is usually behaviour testing, then I'm perfectly fine with this, just need a little simplification.